File: LeetCode_Algorithm/1694(Reformat Phone Number)/1694.py
class Solution:
    def reformatNumber(self, number: str) -> str:
        # sol1 33ms 88% 13.8MB 68%
        number.replace("-", "").replace(" ", "")
        i = 0; ans = []
        while i < len(number):
            if i+4 == len(number):
                ans.append(number[i:i+2])
                i += 2
            elif i+3 <= len(number):
                ans.append(number[i:i+3])
                i += 3
            else:
                ans.append(number[i:i+2])
                i += 2
        return "-".join(ans)

File: LeetCode_Algorithm/1694(Reformat Phone Number)/1694.js
var reformatNumber = function(number) {
  // sol1 58ms 98% 42.3MB 60%
  number = number.replaceAll("-", "").replaceAll(" ", "")
  let i = 0; let ans = []
  while (i < number.length) {
    if (i+4 == number.length) {
      ans.push(number.slice(i, i+2))
      i += 2
    } else if (i+3 <= number.length) {
      ans.push(number.slice(i, i+3))
      i += 3
    } else{
      ans.push(number.slice(i, i+2))
      i += 2
    }
  }
  return ans.join("-")
};